### PR TITLE
Add Cast instance for Nat to String

### PR DESCRIPTION
**Pull request by joom**

### CODE REVIEW

File: libs/prelude/Prelude/Nat.idr [hunk 1]
```diff
@@ -238,6 +238,9 @@ Cast Integer Nat where
 Cast String Nat where
     cast str = cast (the Integer (cast str))
 
+Cast Nat String where
+  cast n = cast (the Integer (cast n))
```
Comment (david-christiansen) [inline libs/prelude/Prelude/Nat.idr]: The indentation here should really match line 239.

**Comment by david-christiansen** [comment]
This is a useful thing! But it seems to introduce some style inconsistencies.

File: libs/prelude/Prelude/Nat.idr (hunk 1 above)
Comment (msmorgan) [inline libs/prelude/Prelude/Nat.idr]: Line 239 is actually inconsistent with the rest of the surrounding lines.

**Comment by joom** [comment]
All implementations indented 2 spaces except the one above. It is fixed now.

**Comment by Melvar** [comment]
Personally, I don’t see the usefulness of a cast to `String` that does the same thing as `show`. As opposed to something like `Cast (List Char) String`, which does something different than `show`.

**Comment by joom** [comment]
Fair point, but there are already `Cast Int String` and `Cast Integer String` instances, which do the same thing as `show`. One would expect a `Cast Nat String` in the same way.

**Comment by melted** [comment]
Thanks!

**Comment by msmorgan** [comment]
Would it be a better idea to implement `Show a` in terms of `Cast a String` throughout the library, establishing a pattern?

**Comment by Melvar** [comment]
That’s not sufficient: `Show` includes `showPrec` to allow correct parenthesization. (It’s what probably most implementations should define, but lots of people miss it and just define `show`.)

**Comment by msmorgan** [comment]
@Melvar I suspected that might be the case.

Hypothetically, is this possible?

```idris

module Prelude.Show
{- ... -}
Cast a String => Show a where
  %overridable
  show = cast

----

module Prelude.Str

Show String where
  show cs = strCons '"' (showLitString (cast cs) "\"")

```

**Pull request merged**